addons/mail/static/src/models/thread_partner_seen_info/thread_partner_seen_info.js

```diff
@@ -15,20 +15,96 @@ function factory(dependencies) {
         /**
          * @override
          */
-        static _createRecordLocalId(data) {
-            return `${this.modelName}_${data.id}`;
+        static _findFunctionFromData(data) {
+            return record => (
+                record.partnerId === data.partnerId &&
+                record.channelId === data.channelId
+            );
+        }
+
+        /**
+         * @private
+         * @returns {mail.partner|undefined}
+         */
+        _computePartner() {
+            if (!this.partnerId) {
+                return [['unlink']];
+            }
+            return [['insert', { id: this.partnerId }]];
+        }
+
+        /**
+         * @private
+         * @returns {mail.thread|undefined}
+         */
+        _computeThread() {
+            if (!this.channelId) {
+                return [['unlink']];
```

Comment:
this should probably just throw (required field)

Reply:
you mean replace the `return` by `throw new Error('ChannelId is required')` ?

Reply:
Ideally the code should absolutely never reach this condition, and crash much earlier if channelId is not set (it should crash at create).
Maybe throw is a bit too extreme, but removing the condition would make the code "cleaner" and already geared towards the future (when we have required field).